meson.build

```diff
@@ -0,0 +1,467 @@
+  libgerbera_args += '-DHAVE_SETLOCALE'
+endif
+
+uuid_dep = dependency('uuid', required : false)
+if not uuid_dep.found()
+  if host_machine.system() == 'linux'
+    error('libuuid is required')
+  else
+    libgerbera_args += '-DBSD_NATIVE_UUID'
+  endif
+endif
+
+# Link to libnsl (Network services library) if it exists. This is something you need on Solaris/OmniOS/Joyent
+nsl_dep = dependency('nsl', required : false)
+# Link to the socket library if it exists. This is something you need on Solaris/OmniOS/Joyent
+socket_dep = dependency('socket', required : false)
+
+if host_machine.system() == 'solaris'
+  libgerbera_args += '-DSOLARIS'
+elif host_machine.system() != 'linux'
+  libgerbera_args += '-DBSD'
+endif
+
+if get_option('buildtype') != 'debug'
+    libgerbera_args += '-DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_INFO'
+else
+    libgerbera_args += '-DTOMBDEBUG'
+    libgerbera_args += '-DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_DEBUG'
+endif
+
+if get_option('avcodec')
+  avcodec_dep = dependency('libavcodec')
+  avfilter_dep = dependency('libavfilter')
+  avformat_dep = dependency('libavformat')
+  avutil_dep = dependency('libavutil')
+  libgerbera_args += '-DHAVE_FFMPEG'
+  if cpp.has_member('struct AVStream', 'codecpar', prefix : '#include <libavformat/avformat.h>', dependencies : avformat_dep)
+    libgerbera_args += '-DHAVE_AVSTREAM_CODECPAR'
+  endif
+else
+  avcodec_dep = declare_dependency()
+  avfilter_dep = declare_dependency()
+  avformat_dep = declare_dependency()
+  avutil_dep = declare_dependency()
+endif
+
+if get_option('curl')
+  curl_dep = dependency('libcurl')
+  libgerbera_args += '-DHAVE_CURL'
+  libgerbera_args += '-DONLINE_SERVICES'
+  libgerbera_args += '-DATRAILERS'
+  libgerbera_args += '-DSOPCAST'
+else
+  curl_dep = declare_dependency()
+endif
+
+if get_option('exif')
+  exif_dep = dependency('libexif')
+  libgerbera_args += '-DHAVE_LIBEXIF'
+else
+  exif_dep = declare_dependency()
+endif
+
+if get_option('exiv2')
+  exiv2_dep = dependency('exiv2')
+  libgerbera_args += '-DHAVE_EXIV2'
+  if exiv2_dep.version().version_compare('<0.28')
+    libgerbera_args += '-D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR'
+  endif
+else
+  exiv2_dep = declare_dependency()
+endif
+
+if get_option('ffmpegthumbnailer')
+  ffmpegthumbnailer_dep = dependency('libffmpegthumbnailer')
+  libgerbera_args += '-DHAVE_FFMPEGTHUMBNAILER'
+else
+  ffmpegthumbnailer_dep = declare_dependency()
+endif
+
+if get_option('inotify')
+  if host_machine.system() == 'linux'
+    libgerbera_args += '-DHAVE_INOTIFY'
+  else
+    error('inotify is Linux only. Please disable.')
+  endif
+endif
+
+if get_option('js')
+  duktape_dep = dependency('duktape')
+  libgerbera_args += '-DHAVE_JS'
+else
+  duktape_dep = declare_dependency()
+endif
+
+if get_option('lastfm')
+  lastfm_dep = dependency('lastfm')
+  libgerbera_args += '-DHAVE_LASTFMLIB'
+else
+  lastfm_dep = declare_dependency()
+endif
+
+if get_option('magic')
+  magic_dep = cpp.find_library('magic')
```

Comment:
The standard file utility and its accompanying libmagic, provides libmagic.pc so dependency() should work here.

Reply:
AFAIK that's only for newer versions. Older versions don't have it.

Reply:
Oh hmm, darn it. Indeed: https://github.com/file/file/commit/2f8ae1e99e89cde6ef6b6d2b5b01480637809aba

My system does have it... :)

In that case you can do a fallback 

```
if get_option('magic') 
    magic_dep = dependency('libmagic', required: false)
    if not magic_dep.found()
        magic_dep = cpp.find_library('magic')
    endif
endif
```

But using feature options is a bit more boilerplate.

Reply:
```
magic_dep = dependency('', required: false)
if not get_option('magic').disabled() 
    magic_dep = dependency('libmagic', required: false)
    if not magic_dep.found()
        magic_dep = cpp.find_library('magic', required: get_option('magic'))
    endif
endif
```

Actually hmm, not so much boilerplate after all...

Reply:
OTOH this project tracks very recent library versions. It's probably easier to make version 39 the minimum.